### PR TITLE
Update pfs start command

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description
- Change to waitress-serve to start waitress in cli, since it's officially recommended.
- Use subprocess to start pfs in detach mode since it provided a more powerful method to manage child process. Using os.spawnv will start pfs failed sometimes. Root cause need more investigations.
I think one possibility is that user doesn't stop pfs when uninstall promptflow. Then user install new promptflow and it will start pfs failed which pollute python virtual environment.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_service/entry.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import logging
 import os
 import platform
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_service/entry.py:8]: add more PR description on what you did in his PR

File: src/promptflow/promptflow/_sdk/_service/entry.py [hunk 2]
```diff
@@ -110,22 +112,13 @@ def validate_port(port, force_start):
...
-            sys.executable,
-            "-m",
-            "waitress",
-            "--host",
-            "127.0.0.1",
-            f"--port={port}",
-            "--call",
-            "promptflow._sdk._service.entry:get_app",
-        ]
+        cmd = ["waitress-serve", f"--listen=127.0.0.1:{port}", "promptflow._sdk._service.entry:get_app"]
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_service/entry.py:115]: I remember msi cannot find waitress-serve command?

Comment (wangchao1230) [reply src/promptflow/promptflow/_sdk/_service/entry.py:115]: We need to have a way to run test on msi in ci

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/promptflow/_sdk/_service/entry.py (hunk 1 above)
Comment (YingChen1996) [reply src/promptflow/promptflow/_sdk/_service/entry.py:8]: added

**Review by lalala123123 — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

**Pull request merged**